### ISSUE TITLE
Editing a clip in the side window does not actually save

### ISSUE THREAD

**Issue by mega-joe** [IndieGoodies/PastePal#59]
**Describe the bug**
Editing a clip in the side window only saves if using Copy as Plain Text. Normal Copy still copies the unedited clip.

**To Reproduce**
1. Copy a full phone number, eg: +1 (212) 555-1212
2. Open the PastePal side window
3. Click on the gear and select Edit
4. In the "Edit" popup, change the number to: 2125551212
5. Press Save
6. In the side window, it correctly shows the edited text. And if you select "Detail" everything looks correct.
7. Select Copy
8. Paste, and you get the original unedited clip (i.e. +1 (212) 555-1212)
9. Select Copy as Plain Text
10. Paste, and you get the edited text (i.e. 2125551212)

So it DOES edit the text, and looks edited, but ONLY if you copy as plain text. It still ALSO preserves the unedited version, even though you can't see it in the side window, or edit the original version anymore.

**Expected behavior**
Editing the clip should commit the change and copy/paste should be with the edited clip, as the clip is shown in the clip history

**Screenshots**
Can provide if needed

**Desktop (please complete the following information):**
 - Mac OS Sonoma 14.5
 - Browser Safari
 - Version 17.5

PastePal Version 2.15.8 (173)

**Additional context**
None

**Comment by onmyway133** [comment]
@mega-joe Hi, thanks for the feedback. When we edit a text (mostly rich text), PastePal updates the plain text version only. In PastePal every content has a plain text version (for image, it is the OCR content of the image if detected).

In latest release 2.15.9, PastePal turns the edit rich text to text now, so you will get the expected result. Please give it a try

**Comment by mega-joe** [comment]
Thanks, the updated version makes more sense and works as expected now.

Someday it might be nice to edit the rich text, as rich text, and keep the edits as rich text, but given that the edit window is only a plain text editor right now, it makes sense that it would convert the clip to plain text and preserve the edits. You would need a rich text editor for that to make sense. Feature request 😄 ??

Anyway, thank you for the quick fix. I've used both Paste and Pastebot for years, and I'm much happier and more productive now that I found PastePal. Keep up the great work!

**Comment by onmyway133** [comment]
@mega-joe good idea, allow editing rich text is also in my plan. It will be a bit hard as there are various rich text format, and also html text from web page is also rich text for now. Rich text can also contain attachment, like images and other formatting as well, imagine Microsoft Word, ...
